slam-libraries/viam-orb-slam3/orbslam_server_v1_main.cc

```diff
@@ -159,6 +162,10 @@ int main(int argc, char **argv) {
         slamService.StartSaveAtlasAsOsa(SLAM.get());
         slamService.ProcessDataOnline(SLAM.get());
         slamService.StopSaveAtlasAsOsa();
+        while (viam::b_continue_session) {
+            this_thread::sleep_for(chrono::microseconds(
+                viam::checkForShutdownIntervalMicroseconds));
+        }
```

Comment:
[q] Why did you add this? I would think that in the case of `ProcessDataOnline` finishing, the user already tried to kill the program and `b_continue_session` would be `false` anyways?

Reply:
leftovers from some debugging that I forgot about, ty for catching